notation login CLI (#223)
notaryproject/notation: specs/notation-cli.md

   certificate, cert  Manage certificates used for verification
   key                Manage keys used for signing
   list, ls           List signatures from remote

>    login              Provide credentials for authenticated registry operations   

Reviewer: ```suggestion
   login              Login to a registry with credentials for authenticated registry operations
```

Author: Saying login, where the parameter says login feels redundant. I was just trying to find a synonym verb.

   --help, -h                  show help (default: false)
```

## login

```console
notation login --help
NAME:

>    notation login - Provides credentials for authenticated registry operations

Reviewer: ```suggestion
   notation login - Login to a registry with credentials for authenticated registry operations
```

USAGE:
   notation login [options] [server]

OPTIONS:
   --username value, -u value    Username for registry operations

>    --password value, -p value    Password for registry operations

Reviewer 2: How about environment variables associated with `-u` and `-p` like `NOTATION_USERNAME` and `NOTATION_PASSWORD`?

Reviewer: ```suggestion
   --username value, -u value    Username for registry authentication
   --password value, -p value    Password for registry authentication
```

Reviewer: Should we also support `--password-stdin` like [docker cli](https://docs.docker.com/engine/reference/commandline/login/#provide-a-password-using-stdin) does?

   --help, -h                    Show help (default: false)

POSITIONAL
  <server>                       The registry URL for authentication

GLOBAL ARGUMENTS

>    --plain-http                  Registry access via plain HTTP (default: false)

Reviewer: @shizhMSFT, what is `--insecure` used for, and how common is the usage for `--ca-file` (I assume provides custom TLS trust store).

>    --plain-http                  Registry access via plain HTTP (default: false)

Reviewer 2: `--insecure` is to skip the TLS cert check. Since `notation` is security oriented, we should not provide this functionality.

`--ca-file` is to specify the root CA cert to registry access. See https://github.com/oras-project/oras/issues/217

>    --plain-http                  Registry access via plain HTTP (default: false)

Author: @shizhMSFT, I was thinking this would be done with the `insecure registries` option in config.  
Does that still exist? 
Seems calling `--insecure` is too tedious. For the scenarios/instances where https isn't supported, such as IOT environments, I just assumed the config would solve the problem and not something that's done ad-hoc requiring the cli to support it.
